blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/IncomingMetricMetadataAnalyzer.java

```diff
@@ -95,10 +92,15 @@ private IncomingMetricException checkMeta(Locator locator, String key, String in
         }
 
         List<IncomingMetricException> problems = new ArrayList<IncomingMetricException>();
+        IncomingMetricException typeProblem = null;
+        IncomingMetricException unitProblem = null;
 
-        IncomingMetricException typeProblem = checkMeta(metric.getLocator(), MetricMetadata.TYPE.name().toLowerCase(),
-                metric.getDataType().toString());
-        IncomingMetricException unitProblem = checkMeta(metric.getLocator(), MetricMetadata.UNIT.name().toLowerCase(),
+        if (metric.getDataType() != DataType.NUMERIC) {
+            typeProblem = checkMeta(metric.getLocator(), MetricMetadata.TYPE.name().toLowerCase(),
+                    metric.getDataType().toString());
+        }
+
+        unitProblem = checkMeta(metric.getLocator(), MetricMetadata.UNIT.name().toLowerCase(),
```

Comment:
NIT: Revert this change to have `IncomingMetricException unitProblem = checkMeta(...`